src/arch/x86_64/CodeGen.zig

```diff
@@ -2576,7 +2586,7 @@ fn airPtrElemVal(self: *Self, inst: Air.Inst.Index) !void {
     const index_ty = self.air.typeOf(bin_op.rhs);
     const index = try self.resolveInst(bin_op.rhs);
     const index_lock: ?RegisterLock = switch (index) {
-        .register => |reg| self.register_manager.lockRegAssumeUnused(reg),
+        .register => |reg| self.register_manager.lockReg(reg),
```

Comment:
Why can this not be `lockRegAssumeUnused`? Not being able to do `lockRegAssumeUnused` for an operand in a "top-level" function (those that start with `air...`) seems suspicious to me.